eclipse-collections-api/src/main/java/org/eclipse/collections/api/factory/list/Lists.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2017 Goldman Sachs and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v. 1.0 which accompany this distribution.
+ * The Eclipse Public License is available at http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ */
+
+package org.eclipse.collections.api.factory.list;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.ServiceLoader;
+import java.util.stream.Collectors;
+
+import org.eclipse.collections.api.block.function.Function;
+
+/**
+ * This class should be used to create instances of MutableList, ImmutableList and FixedSizeList
+ * <p>
+ * Mutable Examples:
+ *
+ * <pre>
+ * MutableList&lt;String&gt; emptyList = Lists.mutable.empty();
+ * MutableList&lt;String&gt; listWith = Lists.mutable.with("a", "b", "c");
+ * MutableList&lt;String&gt; listOf = Lists.mutable.of("a", "b", "c");
+ * </pre>
+ *
+ * Immutable Examples:
+ *
+ * <pre>
+ * ImmutableList&lt;String&gt; emptyList = Lists.immutable.empty();
+ * ImmutableList&lt;String&gt; listWith = Lists.immutable.with("a", "b", "c");
+ * ImmutableList&lt;String&gt; listOf = Lists.immutable.of("a", "b", "c");
+ * </pre>
+ *
+ * FixedSize Examples:
+ *
+ * <pre>
+ * FixedSizeList&lt;String&gt; emptyList = Lists.fixedSize.empty();
+ * FixedSizeList&lt;String&gt; listWith = Lists.fixedSize.with("a", "b", "c");
+ * FixedSizeList&lt;String&gt; listOf = Lists.fixedSize.of("a", "b", "c");
+ * </pre>
+ */
+@SuppressWarnings("ConstantNamingConvention")
+public final class Lists
+{
+    public static final ImmutableListFactory immutable =
+            Lists.loadServiceClass(ImmutableListFactory.class, ThrowingImmutableListFactory::new);
+    public static final MutableListFactory mutable =
+            Lists.loadServiceClass(MutableListFactory.class, ThrowingMutableListFactory::new);
+    public static final FixedSizeListFactory fixedSize =
+            Lists.loadServiceClass(FixedSizeListFactory.class, ThrowingFixedSizeListFactory::new);
```

Comment:
Following @mohrezaei's suggestions, I changed the implementation of loadServiceClass to use a fallback implementation. Now it will never fail at class loading time. The exception message will be the same, but the stack trace will look a little more understandable.